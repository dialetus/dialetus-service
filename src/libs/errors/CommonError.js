class CommonError extends Error {
  constructor({ code, message, statusCode, originalError }) {
    super(message);

    this.name = this.constructor.name;
    this.statusCode = statusCode;
    this.code = code;
    this.originalError = originalError;

    Error.captureStackTrace(this, this.constructor.name);
  }
}

export { CommonError };
