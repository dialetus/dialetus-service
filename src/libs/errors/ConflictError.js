import { CommonError } from './CommonError';

class ConflictError extends CommonError {
  constructor(message = '', originalError) {
    const code = 'ConflictError';
    const statusCode = 409;
    super({ code, statusCode, message, originalError });
  }
}

export { ConflictError };
