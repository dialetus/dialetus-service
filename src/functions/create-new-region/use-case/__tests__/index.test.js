/* eslint-disable max-lines */
import { createNewRegion } from '../index';
import { ConflictError } from '../../../../libs/errors/ConflictError';

describe('# Functions - createNewRegion - use-case', () => {
  it('## should save a new region with success', async () => {
    const findRegionByUF = async (tableName, uf) => {
      return false;
    };

    const dynamoDBSaveItem = async (tableName, data) => {
      return data;
    };
    const tableName = 'mocked-table-data';

    const data = {
      uf: 'ba',
      label: 'Bahia',
      draft: true,
    };

    const newRegion = await createNewRegion(data, {
      dynamoDBSaveItem,
      findRegionByUF,
      ConflictError,
      tableName,
    });
    expect(typeof newRegion).toBe('object');
    expect(newRegion).toMatchObject({ label: 'Bahia', uf: 'BA', draft: true });
  });

  it('## should return an error when try save new region because label is empty', async () => {
    const findRegionByUF = async (tableName, uf) => {
      return false;
    };

    const dynamoDBSaveItem = async (tableName, data) => {
      if (data && !data.label) {
        const error = new Error('One or more parameter values are not valid.');
        error.code = 'ValidationException';
        error.statusCode = 400;
        throw error;
      }
      return data;
    };
    const tableName = 'mocked-table-data';

    const data = {
      uf: 'ba',
      draft: true,
    };
    try {
      await createNewRegion(data, { dynamoDBSaveItem, findRegionByUF, ConflictError, tableName });
    } catch (error) {
      expect(typeof error).toBe('object');
      expect(error).toMatchSnapshot();
    }
  });

  it('## should return an error when try save new region because the UF exist in the table', async () => {
    const findRegionByUF = async (tableName, uf) => {
      return uf.toUpperCase() === 'BA';
    };

    const dynamoDBSaveItem = async (tableName, data) => {
      if (data && !data.label) {
        const error = new Error('One or more parameter values are not valid.');
        error.code = 'ValidationException';
        error.statusCode = 400;
        throw error;
      }
      return data;
    };
    const tableName = 'mocked-table-data';

    const data = {
      uf: 'ba',
      label: 'Bahia',
      draft: 'false',
    };

    try {
      await createNewRegion(data, { dynamoDBSaveItem, findRegionByUF, ConflictError, tableName });
    } catch (error) {
      console.log('error', error);
      expect(error).toMatchSnapshot();
    }
  });
});
