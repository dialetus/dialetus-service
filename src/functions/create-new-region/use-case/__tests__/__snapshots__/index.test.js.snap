// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Functions - createNewRegion - use-case ## should return an error when try save new region because label is empty 1`] = `[Error: One or more parameter values are not valid.]`;

exports[`# Functions - createNewRegion - use-case ## should return an error when try save new region because the UF exist in the table 1`] = `[ConflictError: The UF 'BA' exists in table]`;
